target/scripts/startup/setup-stack.sh

```diff
@@ -1515,6 +1515,27 @@ function _setup_security_stack
       "s|(log_level).*|\1 = ${AMAVIS_LOGLEVEL};|g" \
       /etc/amavis/conf.d/49-docker-mailserver
   fi
+
+  # Per user storage encryption
+  if [[ ${ENABLE_PER_USER_STORAGE_ENCRYPTION} -eq 1 ]]
+  then
+    mv /etc/dovecot/conf.d/10-mailcrypt.disabled /etc/dovecot/conf.d/10-mailcrypt.conf
+    sedfile -i "s|mail_crypt_curve =.*|mail_crypt_curve = ${PER_USER_STORAGE_ENCRYPTION_CURVE}|" /etc/dovecot/conf.d/10-mailcrypt.conf
+    sed -i -e '/\!include auth-passwdfile\.inc/s/^/#/' /etc/dovecot/conf.d/10-auth.conf # comment out old inc
+    sed -i -e '/#\!include auth-passwdfile-mailcrypt\.inc/s/^#//' /etc/dovecot/conf.d/10-auth.conf # uncomment mailcrypt inc
+    sedfile -i "s|scheme={SCHEME}|scheme=${PER_USER_STORAGE_ENCRYPTION_SCHEME}|" /etc/dovecot/conf.d/auth-passwdfile-mailcrypt.inc
+    _notify 'inf' 'Per user storage encryption enabled'
+  else # Ensure it's disabled if the user previously turned it on, but now has ENABLE_PER_USER_STORAGE_ENCRYPTION=0
+    if [[ -f /etc/dovecot/conf.d/10-mailcrypt.conf ]]
+    then
+      mv /etc/dovecot/conf.d/10-mailcrypt.conf /etc/dovecot/conf.d/10-mailcrypt.disabled
+      sedfile -i "s|mail_crypt_curve =.*|mail_crypt_curve = secp521r1|" /etc/dovecot/conf.d/10-mailcrypt.conf.disabled
+      sed -i -e '/\!include auth-passwdfile-mailcrypt\.inc/s/^/#/' /etc/dovecot/conf.d/10-auth.conf # uncomment out old inc
+      sed -i -e '/#\!include auth-passwdfile\.inc/s/^#//' /etc/dovecot/conf.d/10-auth.conf # comment out mailcrypt inc
+      sedfile -i "s|scheme=.*[[:space:]]u|scheme={SCHEME} u|" /etc/dovecot/conf.d/auth-passwdfile-mailcrypt.inc
+      _notify 'inf' 'Per user storage encryption disabled'
+    fi
+  fi
```

Comment:
These have not converted `sed` lines to be consistent with `|` delimiter instead of `/` as raised in [a previous review comment](https://github.com/docker-mailserver/docker-mailserver/pull/2080#discussion_r703972603).

I am not sure but assume that @casperklein request for `sed -i` applies to `sed -i -e` as well?

Reply:
Yes. All options are passed 1:1 from `sedfile` to `sed`. So it is pretty easy to adapt existing code. Just replace `sed -i` with `sedfile -i`.

Reply:
`sed -i -e` to `sedfile -i -e` then @NorseGaud , and the switch from `/` delimiter to `|` thanks :)

Reply:
`sed -i -e '/#\!include auth-passwdfile\.inc/s/^#//'` with `|` doesn't work

It also doesn't work when I use `sedfile -i -e`

Reply:
It wasn't working earlier using sedfile, but now it is.

However, `sedfile -i -e '|\!include auth-passwdfile\.inc|s|^|#|'` does not work.

Any ideas?

Reply:
(I'm experiencing some odd Github behaviour/bugs), @NorseGaud latest comment isn't showing anymore unless it was deleted. I tried to respond via selecting some text and the `...` menu for quote reply which failed to load. Refreshed page and the comment is gone.

No new commits pushed, so sharing the answer here:

`'/#\!include auth-passwdfile\.inc/s|^#||'`

AFAIK, the `|` delimiter only works with the replacement `s` syntax? Keeping the `/` for the earlier `-e` expression seemed required.

[I have avoided `-e` in the past for this type of replacement](https://github.com/docker-mailserver/docker-mailserver/blob/0da66ccb34dcaf0663b6b5635ba97f990b761881/.github/workflows/docs-production-deploy.yml#L39), where you can use a regex group result in the same manner though:

`'s|(^\!include auth-passwdfile\.inc)|#\$1|'`

That appears to be just as valid? So perhaps adopt that approach?: `sedfile -i 's|(^\!include auth-passwdfile\.inc)|#\$1|'`

Reply:
Correction:

- Comment/disable: `sedfile -i 's|^\(\!include auth-passwdfile\.inc\)|#\1|'`
- Uncomment/enable: `sedfile -i 's|^#\(\!include auth-passwdfile\.inc\)|\1|'

Reply:
Looks good -- adding now and testing. Thanks!

Reply:
Works, thanks again.